### DISCUSSION TITLE
Question regarding --log-level "error:debug"

### DISCUSSION THREAD

**Discussion by RobinPC** [appium/appium#18789] [Q&A] [answered]
This is how I spawn Appium:

```
this.appiumServer = spawn(`node ${pathToIndexFile}`, [
    '--address', this.host,
    '--port', this.port,
    '--log-level', this.logLevel, // <-- "debug:error" works fine
    '--log', logPath,
    '--log-timestamp',
    '--long-stacktrace',
    '--log-timestamp',
    '--relaxed-security',
    '--use-plugins', ['all']
], {cwd, shell: true});
```

I hook into the `stdout` like this:
```
this.appiumServer.stdout.on('data', (data) => {
    const message = data.toString();
    // ...
});
```

Why is it that when setting the `--log-level` to `"error:debug"`, the information is not piped through `stdout` anymore? Is there a specific reason for this? And what I can do to retrieve output when I am using `--log-level "error:debug"`? Hooking into `stderr`, `stdin` etc. does not work either.

Versions I am using:
```
    "appium": "2.0.0-beta.71",
    "appium-webdriveragent": "4.13.1",
    "appium-xcuitest-driver": "4.24.3",
    "appium-uiautomator2-driver": "^2.26.0"
```

Thanks!

**Comment by jlipps** [comment, marked as answer]
Which output are you expecting when setting the console level to `error`? Are there actually errors in the output? Appium will only log errors with that setting. So unless you have triggered an error somehow, you'll get no output.

**Reply by RobinPC** [reply]
Thanks for the quick reply. The way I initially read it was that the first option goes to stdout and the second goes to stderr, which confused me. This is not the case, and looking at the documentation clears this up. 

I appreciate the help!
